### ISSUE TITLE
Should `Fraction` be stored as a mixed number?

### ISSUE THREAD

**Issue by LemmaEOF** [FabLabsMC/FluidStacks#5]
Not much else to say, really. We're just deciding whether to:
- store just `long numerator` and `long denominator` as we are right now
- store `int whole`, `int numerator`, and `int denominator`
- store `int whole`,` long numerator`, and `long denominator`

**Comment by shedaniel** [comment]
I don’t really have a preference, would go for one that can be optimised to be faster though.

**Comment by Prospector** [comment]
I think 3 ints should be enough. With wholes split off numerators should never be too high.

**Comment by LemmaEOF** [comment]
yeah, and though Alex did worry that LCMs might get too big for an int, that's why we simplify on every op. The worst LCM I can think of reasonably happening is 81,000 (LCM of 1,000 and 81, which is also a multiple of 3, 4, 5, 6, 8, 9, 10, 12, and most other common units folks would want).

**Comment by kvverti** [comment]
Sounds good to me too. At least have the person who makes the changes document all the decisions.

**Comment by Vonr** [comment]
should whole be stored as a long/biginteger so that it can store more fluid?
